### PR TITLE
feat: event availability preloads with user availability

### PR DESCRIPTION
**Pull request by ivancuadra869**

### CODE REVIEW

File: src/pages/EventPage/EventPage.tsx [hunk 1]
```diff
@@ -88,34 +91,48 @@ export default function EventPage({
...
+                        ),
+                        preloadData:
+                            Object.values(userRecord.availability).length === 0
+                                ? zeroState
+                                : createPreloadArray(
+                                      tempYTimes,
+                                      formatedTempXDays,
+                                      userRecord.availability
+                                  ),
+                        zeroState,
```
Comment (jesc0bar) [inline src/pages/EventPage/EventPage.tsx:130]: Is this still needed? It looks like preloadData will populate a zeroState if no user data is available.

File: src/pages/EventPage/EventPage.tsx [hunk 2]
```diff
@@ -88,34 +91,48 @@ export default function EventPage({
...
-                });
-            },
-        });
-    }, []);
+                    );
+                    const formatedTempXDays = formatXDays(tempXDays);
+                    const zeroState = createZeroStateArray(
+                        LABELS.yLabels.length,
+                        LABELS.xLabels.length
+                    );
```
Comment (jesc0bar) [inline src/pages/EventPage/EventPage.tsx:112]: Should the zeroState be sized to the user defaults here? Was there an issue with how they were written before? If possible I it should be consistent with how we're populating the preloadData from createPreloadArray.

**Review by jesc0bar — APPROVED** [review]
The feature is functional. Good Job! I left comments which discuss changes that could be done here or in a future refactor.

File: src/pages/EventPage/EventPage.tsx (hunk 1 above)
Comment (ivancuadra869) [reply src/pages/EventPage/EventPage.tsx:130]: The zeroState here could be used in the future, for instance in a clear button.

File: src/pages/EventPage/EventPage.tsx (hunk 2 above)
Comment (ivancuadra869) [reply src/pages/EventPage/EventPage.tsx:112]: And you're right, the zero state size should reflect the group availability, I'll change that.

**Review by ameliaabou — APPROVED** [review]
Everything works on my end, looks good! Now to get the data to populate on the heat map and we're done there! 
Also, how do we plan on handling updating/editing one's availability on the shared event map? It's easy with the personal availability map because we could just overwrite the data, but we can't do that with this map can we?

One thing to note that isn't specifically regarding a file that was edited in this PR though. Not sure if it's only on my end, but when I pulled all the recent stuff to my main and tried to run, I was missing a few packages so I ran npm install and it was having some trouble finding the files. 
I had to change the load order in CreateEventForm.tsx so it looked like this:
```
import TimePicker from 'rc-time-picker';
import moment from 'moment';
import { v4 as uuid } from 'uuid';
```
instead of this:
```
import { v4 as uuid } from 'uuid';
import TimePicker from 'rc-time-picker';
import moment from 'moment';
```
Might have just been an error on my end, but I also know that load order can be weirdly specific at times, and I've definitely encountered this issue before while working on this project!

**Comment by ivancuadra869** [comment]
> Everything works on my end, looks good! Now to get the data to populate on the heat map and we're done there! Also, how do we plan on handling updating/editing one's availability on the shared event map? It's easy with the personal availability map because we could just overwrite the data, but we can't do that with this map can we?
> 
> One thing to note that isn't specifically regarding a file that was edited in this PR though. Not sure if it's only on my end, but when I pulled all the recent stuff to my main and tried to run, I was missing a few packages so I ran npm install and it was having some trouble finding the files. I had to change the load order in CreateEventForm.tsx so it looked like this:
> 
> ```
> import TimePicker from 'rc-time-picker';
> import moment from 'moment';
> import { v4 as uuid } from 'uuid';
> ```
> 
> instead of this:
> 
> ```
> import { v4 as uuid } from 'uuid';
> import TimePicker from 'rc-time-picker';
> import moment from 'moment';
> ```
> 
> Might have just been an error on my end, but I also know that load order can be weirdly specific at times, and I've definitely encountered this issue before while working on this project!

Actually, the data already populates in the heat map. We just have to get the availability to save.

And yeah the npm install was required for Sophia's changes from the previous PR. As far as the ordering though, that's odd

**Comment by navarrojandg** [comment]
I just pushed a commit that involved refactoring the CalendarPage by using useUserRecordContext hook so that we save an additional read to the user record. Other than that working great!

**Review by navarrojandg — APPROVED** [review]
Nice job!

**Pull request merged**